Messaging.md

```diff
@@ -87,33 +97,123 @@ The following DLC-specific types are used throughout the specification. All type
+
+To save space, only one side's POV is included in this message as the other can be derived using `remote_payout = total_collateral - local_payout`.
+
+#### Version 0 `contract_descriptor`
+
+1. type: 42768 (`contract_descriptor_v0`)
+2. data:
+   * [`bigsize`:`num_outcomes`]
+   * [`string`:`outcome_1`]
+   * [`u64`:`payout_1`]
    * ...
-   * [`sha256`:`outcome_n`]
-   * [`u64`:`outcome_n_local_payout`]
+   * [`string`:`outcome_num_outcomes`]
+   * [`u64`:`payout_num_outcomes`]
+
+This type represents an enumerated outcome contract.
+
+#### Version 1 `contract_descriptor`
 
-This type of contract info is a simple enumeration of pairs of the hash of an outcome value with the corresponding payout for the local party.
+1. type: 42784 (`contract_descriptor_v1`)
+2. data:
+   * [`u16`:`num_digits`]
+   * [`payout_function`:`payout_function`]
+   * [`rounding_intervals`:`rounding_intervals`]
+
+This type represents a numeric outcome contract.
 
 ### The `oracle_info` Type
 
-This type contains information about the oracle(s) to be used in executing a DLC, and possibly the outcomes possible if these are not specified in the corresponding `contract_info`.
+This type contains information about the oracles to be used in executing a DLC.
 
 #### Version 0 `oracle_info`
 
 1. type: 42770 (`oracle_info_v0`)
 2. data:
-   * [`x_point`:`oracle_public_key`]
-   * [`x_point`:`oracle_nonce`]
+   * [`oracle_announcement`:`oracle_announcement`]
+
+This type of oracle info is for single-oracle events.
+
+#### Version 1 `oracle_info`
+
+1. type: 42786 (`oracle_info_v1`)
+2. data:
+   * [`u16`:`threshold`]
+   * [`u16`:`num_oracles`]
+   * [`oracle_announcement`:`oracle_announcement_1`]
+   * ...
+   * [`oracle_announcment`:`oracle_announcement_num_oracles`]
+
+This type of oracle info is for multi-oracle events where all oracles are signing messages
+that exactly correspond to the messages being signed by the other oracles.
+
+#### Version 2 `oracle_info`
+
+1. type: 55340 (`oracle_info_v2`)
+2. data:
+   * [`u16`:`threshold`]
+   * [`u16`:`num_oracles`]
+   * [`oracle_announcement`:`oracle_announcement_1`]
+   * ...
+   * [`oracle_announcment`:`oracle_announcement_num_oracles`]
+   * [`oracle_params`:`oracle_params`]
+
+The order of the oracle announcements represents a total ordering of preference on the oracles.
+
+This type of oracle info is for multi-oracle numeric events where there is some allotted
+difference between oracles allowed and specified in `oracle_params`.
+
+### The `oracle_params` Type
+
+This type contains information about the oracles being used beyond their announcements.
+
+#### Version 0 `oracle_params` Type
+
+1. type: 55338 (`oracle_params_v0`)
+2. data
+   * [`u16`:`maxErrorExp`]
+   * [`u16`:`minFailExp`]
+   * [`bool`:`maximize_coverage`]
+
+This type is used when there is a shared error bounds requirement for any set of oracles in a
+multi-oracle numeric outcome DLC with allowed (bounded) error.
```

Comment:
This is probably waiting on #128 but should add a link to explain what this is / how it is used